### PR TITLE
Rename NoneTrait to make it private

### LINKED ISSUE #1105: Rename NoneTrait to make it private
`NoneTrait` is currently hidden from `traits.api` because it is intended for internal use of `Union`.
Renaming it to make it private will allow more flexibility for future changes.

### PR DESCRIPTION
**Pull request by kitchoi**
Closes #1105

**Checklist**
- ~Tests~
- [x] Update API reference (`docs/source/traits_api_reference`): autodoc handles that.
- ~Update User manual (`docs/source/traits_user_manual`)~: never mentioned
- [x] Update type annotation hints in `traits-stubs`~

### CODE REVIEW

File: traits/trait_types.py [hunk 1]
```diff
@@ -3810,7 +3810,7 @@ def as_ctrait(self):
         return self.trait_maker.as_ctrait()
 
 
-class NoneTrait(TraitType):
+class _NoneTrait(TraitType):
```
Comment (mdickinson) [inline traits/trait_types.py:3813]: Could we maybe take the opportunity to note in the docstring that this is primarily for use in the `Either` implementation?

Comment (mdickinson) [reply traits/trait_types.py:3813]: Gah. `Union`, not `Either`, of course.

Comment (kitchoi) [reply traits/trait_types.py:3813]: :+1: Yes it would!

**Comment by kitchoi** [comment]
Appveyor build is complaining about connection to packages.enthought.com, restarting to see if this is a temporary network problem...

**Pull request merged**